node/src/components/consensus/highway_core/finality_detector/rewards.rs

```diff
@@ -22,29 +23,20 @@ pub(crate) fn compute_rewards<C: Context>(state: &State<C>, bhash: &C::Hash) ->
     // The panorama of the payout block: Rewards must only use this panorama, since it defines
     // what everyone who has the block can already see.
     let panorama = &payout_vote.panorama;
-    // The vote that introduced the payout block's parent.
-    let opt_parent_vote = payout_block.parent().map(|h| state.vote(h));
-    // The parent's timestamp, or 0.
-    let parent_time = opt_parent_vote.map_or_else(Timestamp::zero, |vote| vote.timestamp);
-    // Only summits for blocks for which rewards are scheduled between the previous and current
-    // timestamps are rewarded, and only if they are ancestors of the payout block.
-    let range = parent_time..payout_vote.timestamp;
-    let is_ancestor =
-        |bh: &&C::Hash| Some(*bh) == state.find_ancestor(bhash, state.block(bh).height);
     let mut rewards = ValidatorMap::from(vec![0u64; panorama.len()]);
-    for bhash in state.rewards_range(range).filter(is_ancestor).unique() {
-        for (vidx, r) in compute_rewards_for(state, panorama, bhash).enumerate() {
+    let mut prev_block = payout_block;
+    while let Some(prop_hash) = prev_block.parent() {
```

Comment:
Why is it called `prop_*`? What does it stand for?

Reply:
"Proposal": It's the proposed block that we're paying rewards for, if it was finalized. (Or partial rewards even if it wasn't yet.) I'll replace `prop` with `proposal`.

Reply:
Danke schon.